src/sentry/relay/config.py

```diff
@@ -397,6 +400,47 @@ def _filter_option_to_config_setting(flt, setting):
 )
 
 
+class _TransactionThreshold(TypedDict):
+    metric: Optional[str]  # Either 'duration' or 'lcp'
+    threshold: float
+
+
+class _TransactionThresholdConfig(TypedDict):
+    projectThreshold: _TransactionThreshold
+    transactionThresholds: Mapping[str, _TransactionThreshold]
+
+
+def _get_satisfaction_thresholds(project: Project) -> _TransactionThresholdConfig:
+    # Always start with the default threshold, so we do not have to maintain
+    # A separate default in Relay:
+    project_threshold: _TransactionThreshold = {
+        "metric": DEFAULT_THRESHOLD["metric"],
+        "threshold": float(DEFAULT_THRESHOLD["threshold"]),
+    }
+
+    # Apply custom project threshold
+    for i, threshold in enumerate(project.projecttransactionthreshold_set.all()):
+        if i > 0:
+            logger.error("More than one transaction threshold applied for project")
+            break
+        project_threshold = {
+            "metric": TRANSACTION_THRESHOLD_KEYS[threshold.metric],
+            "threshold": threshold.threshold,
+        }
+
+    # Apply transaction-specific override
+    return {
+        "projectThreshold": project_threshold,
+        "transactionThresholds": {
+            threshold.transaction: {
+                "metric": TRANSACTION_THRESHOLD_KEYS[threshold.metric],
+                "threshold": threshold.threshold,
+            }
+            for threshold in project.projecttransactionthresholdoverride_set.all()
+        },
+    }
```

Comment:
@shruthilayaj are `ProjectTransactionThreshold` and `ProjectTransactionThresholdOverride` the only two sources where transaction thresholds can be defined, or do I still need to check the "sentry:apdex_threshold" option as well?

Reply:
@jjbayer Yes, ProjectTransactionThreshold  and ProjectTransactionThresholdOverride are the only two places where transaction thresholds are defined/used from. For a particular transaction, if ProjectTransactionThreshold or ProjectTransactionThresholdOverride aren't specified, we fallback to a default value of 300ms applied on the transaction duration. I do still see some references to `sentry:apdex_threshold`  in the codebase,  I will make a PR today to remove those.

Reply:
Thank you!